pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -436,22 +473,23 @@ def __call__(
 
             Names = np.array(["        "], "c")
             pi = np.zeros(ncon, np.float)
-            rc = np.zeros(nvar + ncon, np.float)
             inform = np.array([-1], np.intc)
             mincw = np.array([0], np.intc)
             miniw = np.array([0], np.intc)
             minrw = np.array([0], np.intc)
-            nS = np.array([0], np.intc)
-            ninf = np.array([0], np.intc)
-            sinf = np.array([0.0], np.float)
 
+            if restartDict is not None:
+                hs = restartDict["hs"]
+                xs = restartDict["xs"]
+                pi = restartDict["pi"]
             # The snopt c interface
             timeA = time.time()
             # fmt: off
-            obj = snopt.snkerc(start, nnCon, nnObj, nnJac, iObj, ObjAdd, ProbNm,
-                               self._userfg_wrap, snopt.snlog, snopt.snlog2, snopt.sqlog, self._snstop,
-                               Acol, indA, locA, bl, bu, Names, hs, xs, pi, rc, inform,
-                               mincw, miniw, minrw, nS, ninf, sinf, cu, iu, ru, cw, iw, rw)
+            hs, xs, pi, rc, inform, mincw, miniw, minrw, nS, ninf, sinf, obj = snopt.snkerc(
+                start, nnCon, nnObj, nnJac, iObj, ObjAdd, ProbNm,
+                self._userfg_wrap, snopt.snlog, snopt.snlog2, snopt.sqlog, self._snstop,
+                Acol, indA, locA, bl, bu, Names, hs, xs, pi, cu, iu, ru, cw, iw, rw,
```

Comment:
This change (and the comment you had above about removing other vars) is because, these variables are actually outputs from `snkerc`, not inputs. As a result, I have moved them from the argument side to the output side in the `snkerc` call. They are also not used, because we already call `snmemb` manually. I opted to not replace these variables with just `_` because I felt it was a bit more explicit in matching the function signature in Fortran, but I'm also okay with replacing them.

Reply:
You can also see in the pyf file that the intents have been more clearly labelled.